src/megaapi.cpp

```diff
@@ -1221,47 +1226,49 @@ long long MegaTransfer::getNotificationNumber() const
     return 0;
 }
 
-MegaError::MegaError(int errorCode)
+MegaError::MegaError(int e)
 {
-    this->errorCode = errorCode;
-    this->value = 0;
+    errorCode = e;
 }
 
-MegaError::MegaError(int errorCode, long long value)
+MegaError::~MegaError()
 {
-    this->errorCode = errorCode;
-    this->value = value;
+
 }
 
-MegaError::MegaError(const MegaError &megaError)
+MegaError* MegaError::copy() const
 {
-	errorCode = megaError.getErrorCode();
-    value = megaError.getValue();
+    return nullptr;
```

Comment:
Since it as allowed to have a MegaError that is not actually a MegaErrorPrivate perhaps we should return a copy here?

Reply:
from MegaTransfer::getLastError

Reply:
Agree

Reply:
Solved